parsec/interfaces/ptg/ptg-compiler/parsec.y

```diff
@@ -526,7 +526,24 @@ body:         BODY_START properties BODY_END
              {
                  jdf_body_t* body = new(jdf_body_t);
                  body->properties = $2;
+                 /**
+                  * Go over the list of propertiues and tag them with the type of the BODY. This
+                  * allows us to protect the generated code to avoid compiler warnings.
+                  */
+                 jdf_expr_t* type_str = jdf_find_property( body->properties, "type", NULL );
```

Comment:
Should the name be more specific than "type", for example "architecture" or "accelerator" or something else?

Reply:
This is what we are using right now. We can change it to something more meaningful but then we need to a larger PR.